incompatibility/apiset_test.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2021 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package incompatibility
+
+import (
+	"strconv"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+var ignoreUnexportedSetAnalysis = cmpopts.IgnoreUnexported(
+	IncompatibilityAnalysis{},
+	ApiSetIncompatibility{},
+	FileIncompatibilityClassificationAnalysis{},
+)
+
+// Counts incompatibility from set analysis by classification
+func countIncompSetAnalysis(setAnalysis *ApiSetIncompatibility) []int32 {
```

Comment:
There are a number of fairly complex test helper functions.  This can make it more difficult to review the tests for correctness, which is paramount as there aren't any tests for the test :-D.   

Can you think of any way to reduce the need for these more complicated test helper classes?  Perhaps there is functionality that already exists in analysis which will allow for some of the aggregation you are doing?

Reply:
Breaking out the call to createReport made this much more readable to me.  Looks great.